### ISSUE TITLE
Topology Background

### ISSUE THREAD

**Issue by andersmonteiro** [mattieserver/netbox-topology-views#70]

**Comment by mattieserver** [comment]
It is possible, I am just not sure how good it will look.

Do you have an example image I could use to test (so you can view how it looks before I make the feature available)

**Comment by andersmonteiro** [comment]
I have a network that connects some states. My idea was to be able to insert a 1080p image or superior (cause of zoom) to insert a map and put the network items on top in overlay.

I don't have a specific image at the moment because I haven't defined it yet, but I'll send it as I imagined

![example](https://user-images.githubusercontent.com/48123693/125962825-2d6abb81-8b9a-4d28-b879-5ec17cfbb0a2.png)

**Comment by mattieserver** [comment]
I will have to check if this will work. In the solution I had now the background stayed the same even when you zoomed in, so the location would not be correct depending on how far you are zoomed in.

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden

________________________________

**Comment by tim427** [comment]
I really like the idea! Would be interesting to actually use a real map (OpenStreetMap or GoogleMaps), based on the location coordinates, already present in the database.

**Comment by nonsbe** [comment]
I suppose the original idea was to create a hierarchy-level based topology drawings. Not topologies based on physical locations.

**Comment by tim427** [comment]
> I suppose the original idea was to create hierarchy-level based topology drawings. Not topologies based on physical locations.

I fully agree with this. But when this route is taken (based on some sort of coordinates on a background), I think it would be cool to use real coordinates on a real map, instead of background.

**Comment by nikor30** [comment]
Yes having stacked background images would be great, open street map would be maybe nice if you have interconnects between sites

**Comment by packe1bb** [comment]
What would be _really_ nice, would be if it integrated with Google Maps and used the device location coordinates documented within Netbox.

**Comment by DonMusto** [comment]
> What would be _really_ nice, would be if it integrated with Google Maps and used the device location coordinates documented within Netbox.

I cannot second this enough!

**Comment by mattieserver** [comment]
I created https://github.com/mattieserver/netbox-geographical-views for this.
Will try to get a first release later this month.

**Comment by dreng** [comment]
Since the project at https://github.com/mattieserver/netbox-geographical-views hasn't been started yet and @drygdryg began working on such a plugin (https://github.com/drygdryg/netbox-plugin-device-map), I suggest to bundle the potential instead of splitting work capacity into several projects. Either we can contribute in @drygdryg's project or he can contribute to our project.

In general, I see some difficulties with the proposed feature. While I tought about a feature like that myself a long time ago, I had something like a floor plan in mind (similar to what Ubiquiti provides in their Unifi Controller). It would be difficult to combine both ideas. The same would apply to different locations and sites where you need to provide a map each.

**Comment by joaolucasmacedo** [comment]
I agree wit you @dreng in everything. I think they would combine both plugins. Or, maybe, use device-map code as a base for geographical views.
Also, I think that topology and a map view are different but complementary things.

**Comment by dbrustad** [comment]
would like to second this request- for my use case I'd like to have the ability to specify a custom field for either a site or a region, which would be a floorplan cad drawing exported to whatever format is easiest and lightweight to render as a background. These floorplan images could just live in the same images directory as the icons. In our abstraction we use regions as floors in a building and sites as rooms. Having the ability to place assets on each map and lock them there is quite powerful. mattieserver is there an option to sponsor this functionality?

so the workflow would be topology -> filter against region "floor 22" and then device role "wireless" to see all access points on that floor overlayed accurately over the floorplan image.

This may require the multiple coordinates issue also come to think of it...

**Comment by dreng** [comment]
@dbrustad Although I can understand your workflow and would map it similarly, this would not be THE solution. There are different intentions for what a map should be used for. We first have to come to a consensus on what should be implemented.

So it is not so much a question of sponsoring, but rather a question of what is to be done in the first place. But apart from that, this is an open project. You can always fork the code and pay people to implement your feature and then create a pull request. It doesn't have to be mattiserver or me. I can only speak for myself, but I wouldn't accept money, because these kinds of hobby projects thrive on participation, not payment.

**Comment by kirillmuravyev** [comment]
Hello All!
I dont see much sense with geographical maps as for each device you can store it's address and coordinates. I understand that it can look pretty and nice but what is real use of it?

What makes more sense for me, as a network engineer, and what is aligned with Netbox philosophy is physical location of devices on the site plan.
If you can upload your site plan as a background and fix devices on that plan - that is what is really useful, especially for local IT guys as they should somehow know where each device is located (switches, routers, panels, etc.)

**Comment by mscrnt** [comment]
Hi everyone,

Just curious if there is any update on this feature. I'd really love to get a background image of my site plan added to the topology like what was mentioned above.

**Comment by mattieserver** [comment]
This is currently not planned.
Since we have no way to properly map the devices to background image and the location on it. If you would zoom in/out the view would become totally useless.

**Comment by dreng** [comment]
I'd like to add that we haven't got a proper proposal. This thread is more about brain storming so far. I tend to convert it into a discussion. As stated before there are different requirements. We won't implement anything before it's not absolutely clear what to implement.

Regarding mattieservers post, we may be able to work around the problem he described. That said it is absolutely not worth thinking about workarounds if we don't know what the feature request is actually.

**Comment by dreng** [comment]
Since the proposal is too vague and it has become obvious in the course of the thread that there are different ideas on how to implement it, I am converting this thread into a discussion. We can always create a feature request again when we have precisely defined what should actually be implemented.
